Firestore/core/test/firebase/firestore/remote/datastore_test.mm

```diff
@@ -113,6 +111,52 @@ void Shutdown() {
             "x-google-service: service 2\n");
 }
 
+TEST_F(DatastoreTest, CommitMutationsAuthFailure) {
+  credentials.FailGetToken();
+
+  __block NSError* resulting_error = nil;
+  datastore->CommitMutations(@[], ^(NSError* _Nullable error) {
+    resulting_error = error;
+  });
+  worker_queue.EnqueueBlocking([] {});
+  EXPECT_NE(resulting_error, nil);
+}
+
+TEST_F(DatastoreTest, LookupDocumentsAuthFailure) {
+  credentials.FailGetToken();
+
+  __block NSError* resulting_error = nil;
+  datastore->LookupDocuments(
+      {}, ^(NSArray<FSTMaybeDocument*>* docs, NSError* _Nullable error) {
+        resulting_error = error;
+      });
+  worker_queue.EnqueueBlocking([] {});
+  EXPECT_NE(resulting_error, nil);
+}
+
+/*
+TEST_F(DatastoreTest, AuthWhenDatastoreHasBeenShutDown) {
```

Comment:
This will currently fail; I left it mainly for discussion. I don't know if it can be an issue -- it would depend on how likely it is for `Datastore` to be shut down but not destroyed, so that Auth has a chance to invoke its callback in-between.